packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -699,6 +717,10 @@ export function TransferPanel() {
 
       // show a delay in case of SCW because tx is executed in an external app
       if (isSmartContractWallet) {
+        if (isDepositMode && areSenderAndDestinationAddressesEqual) {
+          await confirmCustomDestinationAddressForSCWallets()
```

Comment:
we should also show the dialog for ERC20 deposits from SCW to same destination address on destination chain